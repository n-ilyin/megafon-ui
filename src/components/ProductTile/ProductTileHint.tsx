import * as React from 'react';
import { cnCreate } from '../../utils/cn';
import './style/ProductTileHint.less';
import { IProductTileHint } from './types';
import { productTileHintProps } from './propTypes';

const cn = cnCreate('mfui-product-tile-hint');
class ProductTileHint extends React.Component<IProductTileHint> {
    static propTypes = productTileHintProps;
    render() {
        const { title } = this.props;

        return (
            <div className={cn('')}>
                <div className={cn('text')}>
                    <span className={cn('description')}>{title}</span>
                </div>
                <div className={cn('border')}/>
            </div>
        );
    }
}

export default ProductTileHint;
