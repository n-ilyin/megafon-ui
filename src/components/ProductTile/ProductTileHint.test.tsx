import * as React from 'react';
import { shallow } from 'enzyme';
import ProductTileHint from './ProductTileHint';

const props = {
    title: 'Тариф-конструктор от 450 ₽',
};

describe('<ProductTileHint />', () => {
    it('it renders ProductTileHint', () => {
        const wrapper = shallow(<ProductTileHint {...props} />);

        expect(wrapper).toMatchSnapshot();
    });
});
