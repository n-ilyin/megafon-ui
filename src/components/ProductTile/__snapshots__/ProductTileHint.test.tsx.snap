// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTileHint /> it renders ProductTileHint 1`] = `
<div
  className="mfui-product-tile-hint"
>
  <div
    className="mfui-product-tile-hint__text"
  >
    <span
      className="mfui-product-tile-hint__description"
    >
      Тариф-конструктор от 450 ₽
    </span>
  </div>
  <div
    className="mfui-product-tile-hint__border"
  />
</div>
`;
